#! /bin/sh /usr/share/dpatch/dpatch-run
## 15_xawtv_bin_cutoff.dpatch by Krzysztof Burghardt <krzysztof@burghardt.pl>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix application command line to not include ".bin" surfix
## DP: (this is required to work transparently with xawtv wrapper).

@DPATCH@
diff -urNad xawtv-3.95.dfsg.1~/x11/xawtv.c xawtv-3.95.dfsg.1/x11/xawtv.c
--- xawtv-3.95.dfsg.1~/x11/xawtv.c	2003-04-11 11:42:04.000000000 +0200
+++ xawtv-3.95.dfsg.1/x11/xawtv.c	2007-09-26 00:00:57.221442822 +0200
@@ -1579,11 +1579,16 @@
 main(int argc, char *argv[])
 {
     int            i;
+    char *         cut;
     unsigned long  freq;
 
     hello_world("xawtv");
     progname = strdup(argv[0]);
-
+    
+    /* cut off .bin from command name */
+    if ((cut = strstr(argv[0], ".bin")))
+        *cut = '\0';
+    
     /* toplevel */
     XtSetLanguageProc(NULL,NULL,NULL);
     app_shell = XtVaAppInitialize(&app_context, "Xawtv",
